clients/java/src/main/java/io/zeebe/client/impl/BearerTokenCredentialsProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.zeebe.client.CredentialsProvider;
+import java.io.File;
+import java.io.IOException;
+
+public class BearerTokenCredentialsProvider implements CredentialsProvider {
+  private static final Key<String> HEADER_AUTH_KEY =
+      Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
+  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
+  private final String zeebeCredentialsPath;
+
+  public BearerTokenCredentialsProvider(String zeebeCredentialsPath) {
+    if (zeebeCredentialsPath == null
+        || zeebeCredentialsPath.isEmpty()
+        || !new File(zeebeCredentialsPath).exists()) {
```

Comment:
Can we extract `new File(zeebeCredentialsPath)` from the condition? In fact can we extract the whole thing to an `isValidCredentialsPath` or something, that'd be more readable I think.